Slim/Control/Queries.pm

```diff
@@ -4722,8 +4727,7 @@ sub worksQuery {
 	}
 
 	if (defined $libraryID) {
-		$sql .= 'JOIN library_track ON library_track.track = tracks.id ';
-		push @{$w}, 'library_track.library = ?';
+		push @{$w}, 'EXISTS (SELECT * FROM library_album WHERE library_album.album = albums.id AND library_album.library = ?)';
```

Comment:
Would it make sense to `SELECT 1 ...` instead of `SELECT * ...`, as we don't really need the data? Similar to when they say that `SELECT COUNT(1)` was faster than `SELECT COUNT(*)`.